lib/funky/post.rb

```diff
@@ -0,0 +1,19 @@
+module Funky
+  class Post < GraphRootNode
+    # @return [String] the type of post.
+    def type
+      data[:type]
+    end
+
+    # @return [DateTime] the created time of the post.
+    def created_time
+      datetime = data[:created_time]
+      DateTime.parse datetime if datetime
+    end
+
+    # @return [Integer] the number of shares of the post.
+    def count_shares
+      data[:shares] && data[:shares][:count] || 0
```

Comment:
@kangkyu What does it mean when `shares` is not there? That the post has 0 shares? Or that the data is not available?

In the first case, it's okay to default to 0. In the second case, we might be better off leaving `nil`.

FIRST CASE

```ruby
# @return [Integer] the number of shares of the post.
def count_shares
  if data[:shares]
    data[:shares][:count]
  else
    0
  end
end
```

SECOND CASE

```ruby
# @return [Integer, nil] the number of shares of the post if available.
def count_shares
  if data[:shares]
    data[:shares][:count]
  end
end
```

In either case, you should add a test for when shares are not available.

Having the `if data[:shares]` on its own line is helpful to guarantee that code coverage tests every case.